source/developers/concepts/hooks_extension_framework.rst

```diff
@@ -0,0 +1,155 @@
++====================+========================================================================+=============================================================+
+| **Purpose**        | Notify when an action occurs in a specific part of the                 | Alter the application flow control.                         |
+|                    | application.                                                           |                                                             |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+|  **Usage**         | Used to **extend** functionality via signal handlers when an event is  |  Used to intercept and **modify** the data used within a    |
+|                    | triggered.                                                             |  component without directly modifying the application       |
+|                    |                                                                        |  itself.                                                    |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+|  **Definition**    |  Defined using the ``OpenEdxPublicSignal``` class, which               |  Defined using the ``OpenEdxPublicFilter`` class,           |
+|                    |  provides a structured way to define the data and                      |  which provides a way to define the filter function         |
+|                    |  metadata associated with the event.                                   |  and the parameters it should receive.                      |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Implementation** |  Implemented using `Django signals`_, which allow                      |  Implemented using an accumulative pipeline mechanism which |
+|                    |  developers to send and receive notifications that an action happened  |  takes a set of arguments and returns a modified set        |
+|                    |  within a Django application.                                          |  to the caller or raises exceptions during                  |
+|                    |                                                                        |  processing.                                                |
++--------------------+------------------------------------------------------------------------+-------------------------------------------------------------+
+| **Use cases**      |  Send an email notification when a user enrolls in a course.           |  Include additional information in an API endpoint response.|
```

Comment:
Could you provide a more detailed example for a filter?

Reply:
I changed this example to "Prevent the enrollment of non-authorized users" https://github.com/openedx/docs.openedx.org/pull/599/commits/c953566eae941823a601d68f712856d501fd5119